File: MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.io.Serializable;
import java.util.ArrayList;
import android.widget.LinearLayout;



import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    LinearLayout containerLayout;
    public class ScheduleItem implements Serializable {
        public String date;
        public String category;
        public int color;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        containerLayout = findViewById(R.id.containerLayout);


        // 전달받은 데이터 가져오기
        Intent intent = getIntent();
        ArrayList<ScheduleItem> scheduleList = (ArrayList<ScheduleItem>) getIntent().getSerializableExtra("scheduleList");

        if (scheduleList == null) {
            scheduleList = new ArrayList<>(); // Null 방지: 빈 리스트로 초기화
        }

        if (scheduleList != null) {
            for (ScheduleItem item : scheduleList) {
                Log.d("MainActivity", "Date: " + item.date + ", Category: " + item.category + ", Color: " + item.color);
            }
        }


        // 반복문 실행 (텍스트 추가 + 색상 적용을 함께 처리)
        for (ScheduleItem item : scheduleList) {
            TextView newTextView = new TextView(this);
            newTextView.setText("날짜: " + item.date + "카테고리: " + item.category);
            newTextView.setTextSize(18);
            newTextView.setPadding(10, 10, 10, 10);
            newTextView.setTextColor(item.color);

            containerLayout.addView(newTextView); // 새로운 TextView 추가
        }


        Button create = findViewById(R.id.Create);
        create.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, NextActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/samyuck/MainActivity.kt
package com.example.samyuck

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageButton
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.io.Serializable
import java.util.Calendar

class MainActivity : AppCompatActivity() {
    //var containerLayout: LinearLayout? = null
    /*
    class ScheduleItem : Serializable {
        var date: String? = null
        var category: String? = null
        var color: Int = 0
    }
    */

    private lateinit var calendarAdapter: CalendarAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //containerLayout = findViewById<LinearLayout>(R.id.containerLayout)
        val addButton = findViewById<ImageButton>(R.id.addButton)//
        addButton.setOnClickListener(View.OnClickListener {
            val intent = Intent(
                this@MainActivity,
                CategoryActivity::class.java
            )
            startActivity(intent)
        })


        setupCalendar()
    }

    private fun setupCalendar() {
        val recyclerView = findViewById<RecyclerView>(R.id.calendarGrid)
        calendarAdapter = CalendarAdapter()

        recyclerView.apply {
            layoutManager = GridLayoutManager(this@MainActivity, 7)
            adapter = calendarAdapter
        }

        // Set current month
        val calendar = Calendar.getInstance()
        calendarAdapter.setCalendarDays(
            calendar.get(Calendar.YEAR),
            calendar.get(Calendar.MONTH)
        )
    }
}